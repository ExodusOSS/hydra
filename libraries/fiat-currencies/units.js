import fiat from '@exodus/fiat'

// Symbol doesn't look right, so we use AED and SAR code directly
const WITH_CODE_INSTEAD_OF_SYMBOL = new Set(['AED', 'SAR'])

// Currencies with these symbols should be displayed on right side
const WITH_RIGHT_SIDE_SYMBOLS = new Set([
  'AED',
  'CHF',
  'CNY',
  'CZK',
  'DKK',
  'GHS',
  'NOK',
  'PLN',
  'SAR',
  'SEK',
  'RUB',
])

// Currencies with symbols that are > 2 in length yet still shown to the left of the balance
const withLeftSymbols = new Set(['RSD'])

const units = Object.fromEntries(
  Object.keys(fiat).map((currency) => {
    const { symbol, label } = fiat[currency]
    const baseProps = {
      symbol: WITH_CODE_INSTEAD_OF_SYMBOL.has(currency) ? currency : symbol,
      label,
      name: currency,
    }
    return [
      currency,
      {
        ...baseProps,
        rightSideSymbol:
          WITH_RIGHT_SIDE_SYMBOLS.has(currency) ||
          (baseProps.symbol.length > 2 && !withLeftSymbols.has(currency)),
      },
    ]
  })
)

export function hasLeftCurrencySymbol(currency) {
  if (!units[currency]) return null
  return !units[currency].rightSideSymbol
}

export default units
