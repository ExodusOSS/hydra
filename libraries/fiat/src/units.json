{
  "AED": {
    "label": "Emirati Dirham",
    "symbol": "د.إ"
  },
  "ARS": {
    "label": "Argentine peso",
    "symbol": "Ar"
  },
  "AUD": {
    "label": "Australian Dollar",
    "symbol": "$"
  },
  "BRL": {
    "label": "Brazilian Real",
    "symbol": "R$"
  },
  "CAD": {
    "label": "Canadian Dollar",
    "symbol": "$"
  },
  "CHF": {
    "label": "Swiss Franc",
    "symbol": "Fr"
  },
  "CLP": {
    "label": "Chilean Peso",
    "symbol": "Ch"
  },
  "CNY": {
    "label": "Chinese Yuan",
    "symbol": "¥"
  },
  "COP": {
    "label": "Colombian peso",
    "symbol": "Co"
  },
  "CZK": {
    "label": "Czech Koruna",
    "symbol": "Kč"
  },
  "DKK": {
    "label": "Danish Krone",
    "symbol": "kr"
  },
  "EUR": {
    "label": "Euro",
    "symbol": "€"
  },
  "GBP": {
    "label": "British Pound",
    "symbol": "£"
  },
  "GHS": {
    "label": "Ghanaian cedi",
    "symbol": "GH₵"
  },
  "HKD": {
    "label": "Hong Kong Dollar",
    "symbol": "$"
  },
  "HUF": {
    "label": "Hungarian Forint",
    "symbol": "Ft"
  },
  "IDR": {
    "label": "Indonesian Rupiah",
    "symbol": "Rp"
  },
  "ILS": {
    "label": "Israeli New Shekel",
    "symbol": "₪"
  },
  "INR": {
    "label": "Indian Rupee",
    "symbol": "₹"
  },
  "JPY": {
    "label": "Japanese Yen",
    "symbol": "¥"
  },
  "KRW": {
    "label": "South Korean Won",
    "symbol": "₩"
  },
  "MXN": {
    "label": "Mexican Peso",
    "symbol": "Mx"
  },
  "MYR": {
    "label": "Malaysian Ringgit",
    "symbol": "RM"
  },
  "NGN": {
    "label": "Nigerian naira",
    "symbol": "₦"
  },
  "NOK": {
    "label": "Norwegian Krone",
    "symbol": "kr"
  },
  "NZD": {
    "label": "New Zealand Dollar",
    "symbol": "$"
  },
  "PHP": {
    "label": "Philippine Peso",
    "symbol": "₱"
  },
  "PKR": {
    "label": "Pakistani Rupee",
    "symbol": "₨"
  },
  "PLN": {
    "label": "Polish Złoty",
    "symbol": "zł"
  },
  "RSD": {
    "label": "Serbian Dinar",
    "symbol": "din"
  },
  "RUB": {
    "label": "Russian Ruble",
    "symbol": "₽"
  },
  "SAR": {
    "label": "Saudi Riyal",
    "symbol": "ر.س"
  },
  "SEK": {
    "label": "Swedish Krona",
    "symbol": "kr"
  },
  "SGD": {
    "label": "Singapore Dollar",
    "symbol": "$"
  },
  "THB": {
    "label": "Thai Baht",
    "symbol": "฿"
  },
  "TRY": {
    "label": "Turkish Lira",
    "symbol": "₺"
  },
  "TWD": {
    "label": "New Taiwan Dollar",
    "symbol": "$"
  },
  "UAH": {
    "label": "Ukrainian hryvnia",
    "symbol": "₴"
  },
  "USD": {
    "label": "United States Dollar",
    "symbol": "$"
  },
  "ZAR": {
    "label": "South African Rand",
    "symbol": "R"
  }
}
