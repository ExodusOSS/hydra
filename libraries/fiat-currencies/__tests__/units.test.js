import units, { hasLeftCurrencySymbol } from '../units.js'

describe('units', () => {
  test('object of units', () => {
    expect(units).toMatchSnapshot()
  })
  test('hasLeftCurrencySymbol CHF', () => {
    expect(hasLeftCurrencySymbol('CHF')).toEqual(false)
  })
  test('hasLeftCurrencySymbol USD', () => {
    expect(hasLeftCurrencySymbol('USD')).toEqual(true)
  })
  test('hasLeftCurrencySymbol undefined', () => {
    expect(hasLeftCurrencySymbol('ASDFASD')).toEqual(null)
  })
})
