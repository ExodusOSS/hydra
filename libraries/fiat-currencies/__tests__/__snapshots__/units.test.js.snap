// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`units object of units 1`] = `
{
  "AED": {
    "label": "Emirati Dirham",
    "name": "AED",
    "rightSideSymbol": true,
    "symbol": "AED",
  },
  "ARS": {
    "label": "Argentine peso",
    "name": "ARS",
    "rightSideSymbol": false,
    "symbol": "Ar",
  },
  "AUD": {
    "label": "Australian Dollar",
    "name": "AUD",
    "rightSideSymbol": false,
    "symbol": "$",
  },
  "BRL": {
    "label": "Brazilian Real",
    "name": "BRL",
    "rightSideSymbol": false,
    "symbol": "R$",
  },
  "CAD": {
    "label": "Canadian Dollar",
    "name": "CAD",
    "rightSideSymbol": false,
    "symbol": "$",
  },
  "CHF": {
    "label": "Swiss Franc",
    "name": "CHF",
    "rightSideSymbol": true,
    "symbol": "Fr",
  },
  "CLP": {
    "label": "Chilean Peso",
    "name": "CLP",
    "rightSideSymbol": false,
    "symbol": "Ch",
  },
  "CNY": {
    "label": "Chinese Yuan",
    "name": "CNY",
    "rightSideSymbol": true,
    "symbol": "¥",
  },
  "COP": {
    "label": "Colombian peso",
    "name": "COP",
    "rightSideSymbol": false,
    "symbol": "Co",
  },
  "CZK": {
    "label": "Czech Koruna",
    "name": "CZK",
    "rightSideSymbol": true,
    "symbol": "Kč",
  },
  "DKK": {
    "label": "Danish Krone",
    "name": "DKK",
    "rightSideSymbol": true,
    "symbol": "kr",
  },
  "EUR": {
    "label": "Euro",
    "name": "EUR",
    "rightSideSymbol": false,
    "symbol": "€",
  },
  "GBP": {
    "label": "British Pound",
    "name": "GBP",
    "rightSideSymbol": false,
    "symbol": "£",
  },
  "GHS": {
    "label": "Ghanaian cedi",
    "name": "GHS",
    "rightSideSymbol": true,
    "symbol": "GH₵",
  },
  "HKD": {
    "label": "Hong Kong Dollar",
    "name": "HKD",
    "rightSideSymbol": false,
    "symbol": "$",
  },
  "HUF": {
    "label": "Hungarian Forint",
    "name": "HUF",
    "rightSideSymbol": false,
    "symbol": "Ft",
  },
  "IDR": {
    "label": "Indonesian Rupiah",
    "name": "IDR",
    "rightSideSymbol": false,
    "symbol": "Rp",
  },
  "ILS": {
    "label": "Israeli New Shekel",
    "name": "ILS",
    "rightSideSymbol": false,
    "symbol": "₪",
  },
  "INR": {
    "label": "Indian Rupee",
    "name": "INR",
    "rightSideSymbol": false,
    "symbol": "₹",
  },
  "JPY": {
    "label": "Japanese Yen",
    "name": "JPY",
    "rightSideSymbol": false,
    "symbol": "¥",
  },
  "KRW": {
    "label": "South Korean Won",
    "name": "KRW",
    "rightSideSymbol": false,
    "symbol": "₩",
  },
  "MXN": {
    "label": "Mexican Peso",
    "name": "MXN",
    "rightSideSymbol": false,
    "symbol": "Mx",
  },
  "MYR": {
    "label": "Malaysian Ringgit",
    "name": "MYR",
    "rightSideSymbol": false,
    "symbol": "RM",
  },
  "NGN": {
    "label": "Nigerian naira",
    "name": "NGN",
    "rightSideSymbol": false,
    "symbol": "₦",
  },
  "NOK": {
    "label": "Norwegian Krone",
    "name": "NOK",
    "rightSideSymbol": true,
    "symbol": "kr",
  },
  "NZD": {
    "label": "New Zealand Dollar",
    "name": "NZD",
    "rightSideSymbol": false,
    "symbol": "$",
  },
  "PHP": {
    "label": "Philippine Peso",
    "name": "PHP",
    "rightSideSymbol": false,
    "symbol": "₱",
  },
  "PKR": {
    "label": "Pakistani Rupee",
    "name": "PKR",
    "rightSideSymbol": false,
    "symbol": "₨",
  },
  "PLN": {
    "label": "Polish Złoty",
    "name": "PLN",
    "rightSideSymbol": true,
    "symbol": "zł",
  },
  "RSD": {
    "label": "Serbian Dinar",
    "name": "RSD",
    "rightSideSymbol": false,
    "symbol": "din",
  },
  "RUB": {
    "label": "Russian Ruble",
    "name": "RUB",
    "rightSideSymbol": true,
    "symbol": "₽",
  },
  "SAR": {
    "label": "Saudi Riyal",
    "name": "SAR",
    "rightSideSymbol": true,
    "symbol": "SAR",
  },
  "SEK": {
    "label": "Swedish Krona",
    "name": "SEK",
    "rightSideSymbol": true,
    "symbol": "kr",
  },
  "SGD": {
    "label": "Singapore Dollar",
    "name": "SGD",
    "rightSideSymbol": false,
    "symbol": "$",
  },
  "THB": {
    "label": "Thai Baht",
    "name": "THB",
    "rightSideSymbol": false,
    "symbol": "฿",
  },
  "TRY": {
    "label": "Turkish Lira",
    "name": "TRY",
    "rightSideSymbol": false,
    "symbol": "₺",
  },
  "TWD": {
    "label": "New Taiwan Dollar",
    "name": "TWD",
    "rightSideSymbol": false,
    "symbol": "$",
  },
  "UAH": {
    "label": "Ukrainian hryvnia",
    "name": "UAH",
    "rightSideSymbol": false,
    "symbol": "₴",
  },
  "USD": {
    "label": "United States Dollar",
    "name": "USD",
    "rightSideSymbol": false,
    "symbol": "$",
  },
  "ZAR": {
    "label": "South African Rand",
    "name": "ZAR",
    "rightSideSymbol": false,
    "symbol": "R",
  },
}
`;
